tests/mail/tests.py

```diff
@@ -1114,42 +1114,50 @@ def test_connection_arg(self):
-        self.assertEqual(len(connection.test_outbox), 2)
-        self.assertEqual(connection.test_outbox[0].subject, "Subject1")
-        self.assertEqual(connection.test_outbox[1].subject, "Subject2")
-
-        connection = mail.get_connection("mail.custombackend.EmailBackend")
-        mail_admins("Admin message", "Content", connection=connection)
-        self.assertEqual(mail.outbox, [])
-        self.assertEqual(len(connection.test_outbox), 1)
-        self.assertEqual(connection.test_outbox[0].subject, "[Django] Admin message")
+        with self.subTest("send_mail()"):
+            connection = mail.get_connection("mail.custombackend.EmailBackend")
+            send_mail(
+                "Subject",
+                "Content",
+                "from@example.com",
+                ["to@example.com"],
+                connection=connection,
+            )
+            self.assertEqual(mail.outbox, [])
+            self.assertEqual(len(connection.test_outbox), 1)
+            self.assertEqual(connection.test_outbox[0].subject, "Subject")
+
+        with self.subTest("send_mass_mail()"):
+            connection = mail.get_connection("mail.custombackend.EmailBackend")
+            send_mass_mail(
+                [
+                    ("Subject1", "Content1", "from1@example.com", ["to1@example.com"]),
+                    ("Subject2", "Content2", "from2@example.com", ["to2@example.com"]),
+                ],
+                connection=connection,
+            )
+            self.assertEqual(mail.outbox, [])
+            self.assertEqual(len(connection.test_outbox), 2)
+            self.assertEqual(connection.test_outbox[0].subject, "Subject1")
+            self.assertEqual(connection.test_outbox[1].subject, "Subject2")
+
+        with self.subTest("mail_admins()"):
+            connection = mail.get_connection("mail.custombackend.EmailBackend")
+            mail_admins("Admin message", "Content", connection=connection)
+            self.assertEqual(mail.outbox, [])
+            self.assertEqual(len(connection.test_outbox), 1)
+            self.assertEqual(
+                connection.test_outbox[0].subject, "[Django] Admin message"
+            )
 
-        connection = mail.get_connection("mail.custombackend.EmailBackend")
-        mail_managers("Manager message", "Content", connection=connection)
-        self.assertEqual(mail.outbox, [])
-        self.assertEqual(len(connection.test_outbox), 1)
-        self.assertEqual(connection.test_outbox[0].subject, "[Django] Manager message")
+        with self.subTest("mail_managers()"):
+            connection = mail.get_connection("mail.custombackend.EmailBackend")
+            mail_managers("Manager message", "Content", connection=connection)
+            self.assertEqual(mail.outbox, [])
+            self.assertEqual(len(connection.test_outbox), 1)
+            self.assertEqual(
+                connection.test_outbox[0].subject, "[Django] Manager message"
+            )
```

Comment:
What do you think about 4 test methods instead of a single test method with 4 subtests? They seem to be quite isolated.

Reply:
Good point, separated out :+1: